packages/react-server-dom-webpack/src/ReactFlightClientConfigNodeBundler.js

```diff
@@ -51,13 +62,21 @@ export function resolveClientReference<T>(
     if (!resolvedModuleData) {
       throw new Error(
         'Could not find the module "' +
-          metadata.id +
+          metadata[ID] +
           '" in the React SSR Manifest. ' +
           'This is probably a bug in the React Server Components bundler.',
       );
     }
-    name = metadata.name;
+    name = metadata[NAME];
   }
+
+  const prefix = bundlerConfig.chunkLoading.prefix;
+  const crossOrigin = bundlerConfig.chunkLoading.crossOrigin;
+  const chunks = metadata[CHUNKS];
+  for (let i = 1; i < chunks.length; i += 2) {
+    preinitModulesForSSR(prefix + chunks[i], crossOrigin);
```

Comment:
Like we discussed, this should probably move out to an explicit call from FlightClient that's not part of "resolving". Maybe it's not part of bundler config at all but maybe more tied to the DOM specific config?

Reply:
Addressed this in the latest push. The layering is a little weird because we have an opaque type for ClientReferenceMetadata in the bundler config so when FlightClient calls into some host specific function with the metadata the only file that can unwrap that is the bundler config

The layering is now
bundler config exposes a proxy method that unwraps the metadata type
destination config exposes a method that implements what we do with webpack chunks when preparing the destination (in this case preiniting)
host config exposes the host specific implementation of preinitialization

The bundler config layer is not necessary but for the opaque type and we could do something else like make a de-opaquing identity function or just type casting but this all gets flattened into a single inline call so I think it is reasonable this way